src/Stratis.Bitcoin.Features.Consensus/ConsensusFeature.cs

```diff
@@ -201,7 +198,8 @@ public static IFullNodeBuilder UsePowConsensus(this IFullNodeBuilder fullNodeBui
                     services.AddSingleton<ConsensusStats>();
                     services.AddSingleton<ConsensusSettings>();
                     services.AddSingleton<IConsensusRules, PowConsensusRules>();
-                    services.AddSingleton<IRuleRegistration, PowConsensusRulesRegistration>();
+
+                    fullNodeBuilder.Network.Consensus.Rules = new PowConsensusRulesRegistration().GetRules();
```

Comment:
Can this not be done in the Network itself?

Reply:
As discussed this will be done in following PRs 👍